app/src/main/java/org/p2p/wallet/receive/tokenselect/dialog/SelectReceiveNetworkBottomSheet.kt

```diff
@@ -0,0 +1,116 @@
+package org.p2p.wallet.receive.tokenselect.dialog
+
+import androidx.core.os.bundleOf
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.setFragmentResult
+import androidx.recyclerview.widget.LinearLayoutManager
+import android.os.Bundle
+import android.view.View
+import org.p2p.core.common.DrawableContainer
+import org.p2p.core.common.TextContainer
+import org.p2p.core.token.Token
+import org.p2p.core.utils.Constants
+import org.p2p.uikit.components.finance_block.FinanceBlockCellModel
+import org.p2p.uikit.components.finance_block.financeBlockCellDelegate
+import org.p2p.uikit.components.icon_wrapper.IconWrapperCellModel
+import org.p2p.uikit.components.left_side.LeftSideCellModel
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.uikit.utils.attachAdapter
+import org.p2p.uikit.utils.image.commonCircleImage
+import org.p2p.uikit.utils.text.TextViewCellModel
+import org.p2p.wallet.R
+import org.p2p.wallet.common.adapter.CommonAnyCellAdapter
+import org.p2p.wallet.common.ui.bottomsheet.BaseRecyclerDoneBottomSheet
+import org.p2p.wallet.receive.tokenselect.models.ReceiveNetwork
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.getColorStateListCompat
+import org.p2p.wallet.utils.withArgs
+
+const val ARG_NETWORKS_TOKENS = "ARG_NETWORKS_TOKENS"
+
+class SelectReceiveNetworkBottomSheet : BaseRecyclerDoneBottomSheet() {
+
+    companion object {
+        fun show(
+            fm: FragmentManager,
+            title: String,
+            requestKey: String,
+            resultKey: String,
+            tokensForNetworks: List<Token>
+        ) = SelectReceiveNetworkBottomSheet().withArgs(
+            ARG_TITLE to title,
+            ARG_NETWORKS_TOKENS to tokensForNetworks,
+            ARG_REQUEST_KEY to requestKey,
+            ARG_RESULT_KEY to resultKey
+        ).show(fm, SelectReceiveNetworkBottomSheet::javaClass.name)
+    }
+
+    private val tokenDataForNetworks: List<Token> by args(ARG_NETWORKS_TOKENS)
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_RoundedSmoke
+
+    private val adapter = CommonAnyCellAdapter(
+        financeBlockCellDelegate(inflateListener = { financeBlock ->
+            financeBlock.setOnClickAction { _, item -> onNetworkClick(item) }
+        })
+    )
+
+    private fun onNetworkClick(item: FinanceBlockCellModel) {
+        val network = (item.payload as? ReceiveNetwork) ?: return
```

Comment:
you can create typedPayloadOrNull<ReceiveNetwork> like typedPayload inside FinanceBlockCellModel